f3dex2.s

```diff
@@ -499,6 +502,44 @@ wait_dpc_start_valid:
     mtc0    $2, DPC_END
 f3dzex_0000111C:
     sw      $2, lbl_00F0
+.else // UCODE_METHOD == METHOD_XBUS
+wait_dpc_start_valid:
+    mfc0 $11, DPC_STATUS
+    andi $11, $11, DPC_STATUS_DMA_BUSY | DPC_STATUS_START_VALID 
+    bne $11, $zero, wait_dpc_start_valid
+     sw $zero, lbl_00F0
+    addi $11, $zero, 2
+    mtc0 $11, DPC_STATUS
+    addi $23, $zero, lbl_0BA8
+    mtc0 $23, DPC_START
+    mtc0 $23, DPC_END
+    lw $12, OSTask + OSTask_flags
+    addi $1, $zero, SP_CLR_SIG2 | SP_CLR_SIG1
+    mtc0 $1, SP_STATUS
+    andi $12, $12, 1
+    beqz $12, f3dzex_xbus_0000111C
+     sw $zero, OSTask + OSTask_flags
+    j load_overlay1_init
+     lw $26, lbl_0BF8
+    nop // Bunch of nops here to make it the same size as the fifo code.
```

Comment:
If you are interested in making this more compact, armips supports `.fill`, for example
`.fill 0x40` or `.fill 16 * 4`